File: extensions/reload.py
""" Ensure we can reload aeguir without restarting """
from common import bust_cache, get_config, is_mod
from discord.ext import commands


class Reload(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.command()
    async def reload(self, ctx):
        if ctx.author.bot:
            return
        if not ctx.guild:
            return
        if not is_mod(ctx.author):
            return
        new_val = bust_cache(ctx.guild)
        await ctx.channel.send("Cache index now = {}".format(new_val))


def setup(bot):
    cog = Reload(bot)
    bot.add_cog(cog)

File: extensions/reload.js
const {
  bustCache,
  getConfig,
  isMod,
} = require('../common.js');
const {MessageEmbed} = require('discord.js');

const EMOJIS = ['⏸', '🛑'];

module.exports = {
  setup: (client) => {
    client.on('messageCreate', async (message) => {
      if (message.partial) {
        try {
          await message.fetch();
        } catch (error) {
          console.error(error);
          return;
        }
      }
      if (message.content.toLowerCase() !== 'aeguir reload') {
        return;
      }
      const {member, guild} = message;
      if (!guild) {
        return;
      }
      if (!member) {
        return;
      }
      if (member.partial) {
        try {
          await member.fetch();
        } catch (error) {
          console.error(error);
          return;
        }
      }
      if (member?.user?.bot) {
        return;
      }
      if (!isMod(member)) {
        return;
      }
      const newCacheVal = bustCache(guild);
      await message.reply(`Cache index now = ${newCacheVal}`);
    });
  },
};
